modules/utxo-lib/src/bitgo/keyutil.ts

```diff
@@ -36,3 +37,29 @@ export function privateKeyBufferFromECPair(ecPair: ECPairInterface): Buffer {
 
   return privkey;
 }
+
+/**
+ * Converts an extended key from one network to another by updating its version bytes.
+ *
+ * Handles both public and private keys, allowing conversion between networks like
+ * Bitcoin Mainnet and Testnet.
+ *
+ * @returns The extended key with the updated network version.
+ */
+export function convertExtendedKeyNetwork(extendedKey: string, fromNetwork: Network, targetNetwork: Network): string {
+  if (fromNetwork === targetNetwork) {
+    return extendedKey;
+  }
+
+  try {
+    const decodedData = bs58check.decode(extendedKey);
+    const hdNode = bip32.fromBase58(extendedKey, fromNetwork);
+    const targetVersionBytes = hdNode.isNeutered() ? targetNetwork.bip32.public : targetNetwork.bip32.private;
+    const versionBuffer = Buffer.alloc(4);
+    versionBuffer.writeUInt32BE(targetVersionBytes, 0);
+    versionBuffer.copy(decodedData, 0, 0, 4);
+    return bs58check.encode(decodedData);
+  } catch (error) {
+    throw new Error(`Failed to convert extended key from ${fromNetwork} to ${targetNetwork}: ${error.message}`);
```

Comment:
networks themselves cannot be stringified - you can use `getNetworkName`

the `catch`  will hide the exception details btw (line number) so consider removing it